### DISCUSSION TITLE
Best practices for real time doc editing

### DISCUSSION THREAD

**Discussion by tawjaw** [supabase/supabase#748] [Questions] [answered]
Hello, 

Can anyone share ideas or best practices for building a real-time doc editing? Like google docs for example. 

Thanks

**Comment by kiwicopple** [comment, marked as answer]
Hey @tawjaw - for this you probably want to use something like Yjs: https://github.com/yjs/yjs

Collaborative text editors are quite a complex topic. Google Docs use "Operational Transforms" but CRDTs are looking like they are the way forward: https://josephg.com/blog/crdts-are-the-future/

I'd start with the collaborative editor (like ProseMirror)

**Reply by tawjaw** [reply]
Thanks @kiwicopple 
These references are great!
